lisa/notifiers/text_result.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Any, List, Type, Union
+
+from lisa import notifier, schema
+from lisa.runner import print_results
+from lisa.testsuite import TestResult, TestResultMessage
+from lisa.util import constants
+
+
+class TextResult(notifier.Notifier):
+    """
+    Creating log notifier to dump text formatted results for easier
+    view in editing mode. The original log is complete but too long to
+    check only the summary.
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return constants.NOTIFIER_TEXT_RESULT
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return schema.Notifier
+
+    def _received_message(self, message: TestResultMessage) -> None:
+        if message.is_completed:
+            self.received_messages.append(Union(None, message))
```

Comment:
Raise an error in case the message is different from the ones subscribed : 
```
 if isinstance(message, notifier.TestResultMessage):
  ......    
  else:
      raise LisaException("Received unsubscribed message type")
```